generator/integration_tests/benchmarks/client_benchmark.cc

```diff
@@ -0,0 +1,237 @@
+      google::test::admin::database::v1::Request,
+      google::test::admin::database::v1::Response>>
+  AsyncStreamingReadWrite(google::cloud::CompletionQueue const&,
+                          std::unique_ptr<grpc::ClientContext>) override {
+    return nullptr;
+  }
+
+  Status ExplicitRouting1(
+      grpc::ClientContext&,
+      google::test::admin::database::v1::ExplicitRoutingRequest const&)
+      override {
+    return Status();
+  }
+
+  Status ExplicitRouting2(
+      grpc::ClientContext&,
+      google::test::admin::database::v1::ExplicitRoutingRequest const&)
+      override {
+    return Status();
+  }
+
+  std::unique_ptr<::google::cloud::internal::AsyncStreamingReadRpc<
+      google::test::admin::database::v1::Response>>
+  AsyncStreamingRead(
+      google::cloud::CompletionQueue const&,
+      std::unique_ptr<grpc::ClientContext>,
+      google::test::admin::database::v1::Request const&) override {
+    return nullptr;
+  }
+
+  std::unique_ptr<::google::cloud::internal::AsyncStreamingWriteRpc<
+      google::test::admin::database::v1::Request,
+      google::test::admin::database::v1::Response>>
+  AsyncStreamingWrite(google::cloud::CompletionQueue const&,
+                      std::unique_ptr<grpc::ClientContext>) override {
+    return nullptr;
+  }
+};
+
+std::shared_ptr<GoldenKitchenSinkConnection> MakeTestConnection(
+    std::shared_ptr<GoldenKitchenSinkStub> stub, Options options) {
+  options = GoldenKitchenSinkDefaultOptions(std::move(options));
+  auto background = internal::MakeBackgroundThreadsFactory(options)();
+  return std::make_shared<GoldenKitchenSinkConnectionImpl>(
+      std::move(background), std::move(stub), std::move(options));
+}
+
+template <int N>
+struct ExtraOption {
+  using Type = int;
+};
+
+void BM_ClientRoundTripStubOnly(benchmark::State& state) {
+  auto options = Options{};
+  auto stub = std::make_shared<TestStub>();
+  auto conn = MakeTestConnection(std::move(stub), std::move(options));
+  auto client = GoldenKitchenSinkClient(std::move(conn));
+
+  for (auto _ : state) {
+    auto status = client.DoNothing();
+    benchmark::DoNotOptimize(status);
+  }
+}
+BENCHMARK(BM_ClientRoundTripStubOnly);
+
+void BM_ClientRoundTripMetadata(benchmark::State& state) {
+  auto options = Options{};
+  std::shared_ptr<GoldenKitchenSinkStub> stub = std::make_shared<TestStub>();
+  stub = std::make_shared<GoldenKitchenSinkMetadata>(std::move(stub));
+  auto conn = MakeTestConnection(std::move(stub), std::move(options));
+  auto client = GoldenKitchenSinkClient(std::move(conn));
+
+  for (auto _ : state) {
+    auto status = client.DoNothing();
+    benchmark::DoNotOptimize(status);
+  }
+}
+BENCHMARK(BM_ClientRoundTripMetadata);
+
+void BM_ClientRoundTripLogging(benchmark::State& state) {
+  auto options = Options{};
+  std::shared_ptr<GoldenKitchenSinkStub> stub = std::make_shared<TestStub>();
+  stub = std::make_shared<GoldenKitchenSinkLogging>(
```

Comment:
If you want to make the test look better / different, try:

```cc
google::cloud::LogSink::Instance().set_minimum_severity(google::cloud::Severity::GCP_LS_WARNING);
```

Reply:
```
Run on (96 X 2000 MHz CPU s)
CPU Caches:
  L1 Data 32 KiB (x48)
  L1 Instruction 32 KiB (x48)
  L2 Unified 1024 KiB (x48)
  L3 Unified 39424 KiB (x2)
Load Average: 0.73, 0.29, 0.18
----------------------------------------------------------------------------
Benchmark                                  Time             CPU   Iterations
----------------------------------------------------------------------------
BM_ClientRoundTripStubOnly              2353 ns         2353 ns       297597
BM_ClientRoundTripMetadata              2775 ns         2775 ns       253992
BM_ClientRoundTripLogging               2372 ns         2372 ns       293695
BM_ClientRoundTripTenExtraOptions       4987 ns         4986 ns       140084
```

interesting....